rplugin/python3/deoplete/sources/deoplete_jedi/server.py

```diff
@@ -524,12 +524,17 @@ def completions(self, *args):
         restart the server.
         """
         if self._count > self.max_completion_count:
-            self._count = 0
             self.restart()
 
         self._count += 1
-        stream_write(self._server.stdin, args)
-        return stream_read(self._server.stdout)
+        try:
+            stream_write(self._server.stdin, args)
+            return stream_read(self._server.stdout)
+        except StreamError as exc:
+            log.error('Caught %s during handling completions(%s), '
+                      ' restarting server', exc, args)
```

Comment:
Deoplete's logger discards repetitive messages if they come in too fast, but only if it's the same message.  My main motivation for keeping in-vim messages minimal was actually for deoplete-related development.  I'd often break something and every keystroke made a giant message scroll on the screen.  😳

> Maybe use some kind of limit here?

That's your call.  I never experienced what's described in #56 and am not sure if this is only a once-in-a-while occurrence.

> Do you recommend a way to test issues in this context?

Unfortunately, yes, just throw an error manually for this one.  It could be a random chance throw if you want to emulate it recovering.

Unit testing was too hard to think through for the current server.  If I ever needed a problem to happen in the server, I would just make the server look out for a certain cache key and have it reproduce problems.  That way, it wouldn't be a 100% chance of occurrence during dev.